### ISSUE TITLE
Feature Request: Option to Disable Page Revisioning and Restores

### ISSUE THREAD

**Issue by nh-mike** [BookStackApp/BookStack#320]
Hi all,

I am requesting to have an option in settings to disable page revisions and restoring. While this is a nice feature, it is unnecessary for my use of Bookstack and will lead to a larger database size that I will find necessary, and I am sure that somewhere, someone else might find this useful too. It would also be nice to be able to turn off recent activities, and to have the option to set the redirect to /books when a user logs in.

In short, it would be nice if we could keep our activities and page_revisions tables empty.

Thanks guys.

**Comment by ssddanbrown** [comment]
Hi @nh-mike, Thanks for your suggestion.

I understand your reasoning but I'm a little adverse to making some of the core features conditional (From a project maintainability point of view) without a lot of request for it.

If we added a couple of commands: A 'Clear revisions' command and a 'Clear Activity' command, Would that help? That way you could run those commands periodically on a cron job to keep database sizes down and those features would still have short-term benefits (Such as rolling back a change straight after making a mistake).

**Comment by nh-mike** [comment]
Hi @ssddanbrown, while it is not exactly what I am looking for, we can run with it. I appreciate the reasoning to not do this, and would be happy with a cron accessible command to truncate the tables. Yes, this would be helpful.

Thanks.

**Comment by ssddanbrown** [comment]
Whoops, Looks like I forgot to comment back on this issue after the release of v0.15. Since the above mentioned commands were added in v0.15 ([documentation here](https://www.bookstackapp.com/docs/admin/commands/)) I will close this issue. I'll keep the original request in mind and keep note of anyone else requesting the ability to disable restores or revisions.

**Comment by quaintdev** [comment]
I referred to documentation and it seems there is no way to do this from user interface which is understandable since this is not something one would do frequently.

My issue is I am running Bookstack via docker.  Tried to run clear revision command inside main docker container but it gives error `cannot open input file: artisan`. Where should I execute `php artisan` command inside the container? Any help would be highly appreciated.

Thanks!

**Comment by nh-mike** [comment]
Hi @quaintdev yes, you must run it inside the container.

`docker exec -it bookstack /usr/bin/php /var/www/html/artisan`

Of course followed by any necessary artisan commands. This assumes you have named your container "bookstack".

**Comment by noname1001** [comment]
I couldn't execute the command because artisan is in a different folder.

My bookstack setup is based on the lscr.io/linuxserver/bookstack image.

I wanted to clear the revisions and had to run this command:

`docker exec -it bookstack php /app/www/artisan bookstack:clear-revisions`

**Comment by nh-mike** [comment]
Hi @noname1001

I have not used this on docker myself - by chance do you know the path to artisan within the container? It looks to me like it is stored in /config? If so, you should be able run it from there like so; 
docker exec -it bookstack php /config/www/artisan bookstack:clear-revisions

If not then this is a query for the LinuxServer team.
